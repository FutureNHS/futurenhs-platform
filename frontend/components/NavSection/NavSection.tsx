import React, { useState } from "react";

import styled from "styled-components";

import { ChevronBottomIcon, ChevronTopIcon } from "../Icon";

const Section = styled.section`
  padding-top: 20px;
  div {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: start;
  }
  button {
    border: none;
    background: inherit;
  }
  .chevron {
    outline: inherit;
    border-top: 4px solid transparent;
    border-bottom: 4px solid transparent;
    cursor: pointer;
    :focus {
      background-color: ${({ theme }) => theme.colorNhsukYellow};
      border-bottom: 4px solid ${({ theme }) => theme.colorNhsukBlack};
    }
  }
`;

const NavSection = ({
  title,
  children,
  initiallyOpen = true,
}: {
  title: string;
  children: any;
  initiallyOpen?: boolean;
}) => {
  const [open, setOpen] = useState(initiallyOpen);

  return (
    <Section>
      <div>
        <h4>{title}</h4>
        <button className="chevron" onClick={() => setOpen(!open)}>
          {open ? (
            <ChevronTopIcon title={`Hide ${title}`} />
          ) : (
            <ChevronBottomIcon title={`Show ${title}`} />
          )}
        </button>
      </div>
      {open && children}
    </Section>
  );
};

export default NavSection;
