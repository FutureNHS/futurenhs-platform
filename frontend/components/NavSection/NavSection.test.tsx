import React from "react";

import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";

import theme from "../../lib/fixtures/theme.json";
import NavSection from "./NavSection";

describe("<NavSection/>", () => {
  it("shows children when open (default)", () => {
    const container = render(
      <ThemeProvider theme={theme}>
        <NavSection title="Folders">Look at me! I am children!</NavSection>
      </ThemeProvider>
    );

    expect(container.asFragment()).toMatchSnapshot();
  });

  it("does not render children when closed", () => {
    const container = render(
      <ThemeProvider theme={theme}>
        <NavSection title="Folders" initiallyOpen={false}>
          I should not exist because nav is not open.
        </NavSection>
      </ThemeProvider>
    );

    expect(container.asFragment()).toMatchSnapshot();
  });
});
