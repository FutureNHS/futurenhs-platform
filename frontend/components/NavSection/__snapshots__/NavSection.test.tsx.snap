// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavSection/> does not render children when closed 1`] = `
<DocumentFragment>
  .c0 {
  padding-top: 20px;
}

.c0 div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.c0 button {
  border: none;
  background: inherit;
}

<section
    class="c0"
  >
    <div>
      <h4>
        Folders
      </h4>
      <button>
        <img
          alt="Show Folders"
          src="/public/chevronClosed.svg"
        />
      </button>
    </div>
  </section>
</DocumentFragment>
`;

exports[`<NavSection/> shows children when open (default) 1`] = `
<DocumentFragment>
  .c0 {
  padding-top: 20px;
}

.c0 div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.c0 button {
  border: none;
  background: inherit;
}

<section
    class="c0"
  >
    <div>
      <h4>
        Folders
      </h4>
      <button>
        <img
          alt="Hide Folders"
          src="/public/chevronOpen.svg"
        />
      </button>
    </div>
    Look at me! I am children!
  </section>
</DocumentFragment>
`;
